packages/plugin-ext/src/plugin/scm.ts

```diff
@@ -111,8 +118,8 @@ function compareResourceThemableDecorations(a: theia.SourceControlResourceThemab
         return 1;
     }
 
-    const aPath = typeof a.iconPath === 'string' ? a.iconPath : a.iconPath.fsPath;
-    const bPath = typeof b.iconPath === 'string' ? b.iconPath : b.iconPath.fsPath;
+    const aPath = typeof a.iconPath === 'string' ? a.iconPath : URI.isUri(a.iconPath) ? a.iconPath.fsPath : (a.iconPath as theia.ThemeIcon).id;
```

Comment:
Yes, this can end up comparing a Theme Icon ID to a File System Path. This part was copied from [here](https://github.com/microsoft/vscode/blob/d1aa00acef8b2cfa88621c448b8fd8cd034f60a9/src/vs/workbench/api/common/extHostSCM.ts#LL55C1-L56C1)

As far as I am aware the Theme Icon IDs are codicon IDs. Thus, they should all be in lowercase.